### PR TITLE
Views

### PR DESCRIPTION
**Pull request by eudondian**
Views

Implemented the design from the sneak peek wireframes.
Used methods created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
+  <p class='user_empty-message'>No users yet!</p>
+<% else %>
+  <div class='container'>
+    <% @users.each do |user| %>
+      <div>
+        <div class='avatar'>
+            <%= image_tag(user.photo) %>
+          </div>
+          <div><h2><%= user.name %></h2></div>
+          <div> <p><%= link_to "View Profile", user_path(user) %></p> </div>
```
Comment (levy002) [inline app/views/users/index.html.erb]: - [ ] KIndly make sure to display post counts on user profiles to enhance user experience and engagement. Let's make this improvement to achieve our project milestone successfully.

   - ## PROVIDED DESIGN 
  image

   - ### YOUR DESIGN
 image

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
-
+<h1 class='title'>Welcome to Blog-Haven</h1>
+<div class= "container">
+<% if @user %>
+    <div class='user'>
+      <div class='avatar'>
+        <%= image_tag(@user.photo) %>
+      </div>
+    <div><h2><%= @user.name %></h2></div>
+    <div><p>No of Posts:</p></div>
```
Comment (levy002) [inline app/views/users/show.html.erb]: - [ ] KIndly make sure to display post counts on user profiles to enhance user experience and engagement. Let's make this improvement to achieve our project milestone successfully.

   - ## PROVIDED DESIGN 
  image

   - ### YOUR DESIGN
image

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @eudondian   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
+      <div>
+        <div class='avatar'>
+            <%= image_tag(user.photo) %>
+          </div>
+          <div><h2><%= user.name %></h2></div>
+          <div> <p><%= link_to "Number of posts: x", user_path(user) %></p> </div>
+         
+         
+        
+      </div>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/index.html.erb]: - [ ] As the previous reviewer mentioned, the post count is not showing, and also you are not matching the wireframe given
your design :point_down: 
Screenshot from 2023-07-21 23-28-06
given design :point_down: 
Screenshot from 2023-07-21 23-35-02

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
+      </div>
+      </div>
+  
+<div class="recent-posts">
+  <% if @user.most_recent_posts.any? %>
+    <% @user.most_recent_posts.each do |post| %>
+        <div class='recent-post'>
+          <h3><%= link_to post.title, user_post_path(@user, post), class: 'post-title' %> Post</h3>
+          <p><%= post.text %><br/>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda, quas magni. Iure deserunt voluptates accusantium quam maxime tempore sint veniam laudantium? Aperiam placeat dolor quasi facilis! Dicta quasi soluta fugiat cupiditate facere consequuntur ipsum! Ea beatae, adipisci delectus sunt rerum veniam velit odio sit deleniti deserunt nobis facere nostrum ex! </p>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/show.html.erb]: - [ ] You do not need the dummy text, kindly remove them

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1,12 +1,36 @@
...
+        <%= image_tag(@user.photo) %>
+      </div>
+    <div><h2><%= @user.name %></h2></div>
+    <div><p>Number of Posts: x</p></div>
+    </div>
+    <div class='user-info'>
+      <div><h2>Bio</h2></div>
+      <div><p><%= @user.bio %><br/>Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellat aliquid eius fuga. Nulla reprehenderit, ipsa quam, asperiores itaque totam vel amet iste ducimus pariatur distinctio reiciendis quasi hic, illum sit.Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda, quas magni. Iure deserunt voluptates accusantium quam maxime tempore sint veniam laudantium? Aperiam placeat dolor quasi facilis! Dicta quasi soluta fugiat cupiditate facere consequuntur ipsum! Ea beatae, adipisci delectus sunt rerum veniam velit odio sit deleniti deserunt nobis facere nostrum ex!</p>
+      </div>
+      </div>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/show.html.erb]: - [ ] As the previous reviewer mentioned kindly fix the post count and also match the given wireframe 
in your design :point_down: 
Screenshot from 2023-07-21 23-39-39
given wireframe :point_down: 
Screenshot from 2023-06-14 23-02-12

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @eudondian , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/views/users/index.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/users/index.html.erb]: well resolved! ✔️
Design follows the wireframe though there's not bother applied to design.

File: app/views/users/index.html.erb (hunk 3 above)
Comment (rloterh) [reply app/views/users/index.html.erb]: well resolved! ✔️
Design follows the wireframe though there's not border applied to design.

File: app/views/users/show.html.erb (hunk 4 above)
Comment (rloterh) [reply app/views/users/show.html.erb]: well resolved! ✔️

File: app/views/users/show.html.erb (hunk 5 above)
Comment (rloterh) [reply app/views/users/show.html.erb]: well resolved! ✔️
Design follows the wireframe though there's not border applied to design.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Esther,  

Your project is complete! 🏁 Great job for nailing it right on the first attempt 🥇 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvement to your app design and app functionality!
🎲 Good ReadMe!


## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Generally, it's good idea the subheading text _Author1_ should be replaced with your full name in your ReadMe.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**